lib/src/component_declaration/component_base.dart

```diff
@@ -275,23 +275,24 @@ typedef PropsModifier(Map props);
 abstract class UiProps
     extends Object with MapViewMixin, PropsMapViewMixin, ReactPropsMixin, UbiquitousDomPropsMixin, CssClassPropsMixin
     implements Map {
-  /// Adds an arbitrary prop key-value pair.
-  void addProp(propKey, value) {
+  /// Adds an arbitrary prop key-value pair if [shouldAdd] is true, otherwise, does nothing.
+  void addProp(propKey, value, [bool shouldAdd = true]) {
+    if (!shouldAdd) return;
+
     props[propKey] = value;
   }
 
-  /// Adds a Map of arbitrary props. [props] may be null.
-  void addProps(Map propMap) {
-    if (propMap == null) {
-      return;
-    }
+  /// Adds a Map of arbitrary props if [shouldAdd] is true, otherwise, does nothing.
+  /// [props] may be null.
+  void addProps(Map propMap, [bool shouldAdd = true]) {
+    if (!shouldAdd || propMap == null) return;
 
     props.addAll(propMap);
   }
 
-  /// Allows [modifier] to alter this instance of props.
-  void modifyProps(PropsModifier modifier){
-    if (modifier == null) return;
+  /// Allows [modifier] to alter this instance of props if [shouldModify] is true, otherwise, does nothing.
```

Comment:
s/if [shouldModify] is true/if [shouldModify] is true and [modifier] is not null